File: typescript/src/examples/evm/listTokenBalances.ts
// Usage: pnpm tsx src/examples/evm/listTokenBalances.ts

import { config } from "dotenv";

import { CdpClient } from "../../index.js";

/**
 * This example shows how to list the token balances of an EVM address
 */
async function main() {
  config();

  const cdp = new CdpClient();

  // Get the first page of token balances for barmstrong.eth.
  const address = "0x5b76f5B8fc9D700624F78208132f91AD4e61a1f0";

  const result = await cdp.evm.listTokenBalances({
    address,
    network: "base-sepolia",
  });

  for (const balance of result.balances) {
    console.log("Token contract address:", balance.token.contractAddress);
    console.log("Token balance:", balance.amount.amount);
  }
}

main().catch(console.error);

File: python/cdp/examples/evm/list_token_balances.py
# Usage: uv run python cdp/examples/evm/list_token_balances.py

import asyncio

from dotenv import load_dotenv

from cdp import CdpClient


async def main():
    """Contains main function for the EVM token balances script."""
    load_dotenv()

    async with CdpClient() as cdp:
        address = "0x5b76f5B8fc9D700624F78208132f91AD4e61a1f0"
        network = "base-sepolia"
        token_balances = await cdp.evm.list_token_balances(address, network)

        for balance in token_balances.balances:
            print(
                f"Contract Address: {balance.token.contract_address} Balance: {balance.amount.amount}"
            )


# Run the async function
asyncio.run(main())
